dotnet/Vaas/src/Vaas/Vaas.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.WebSockets;
+using System.Security.Cryptography;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Vaas.Messages;
+using Websocket.Client;
+using Websocket.Client.Exceptions;
+
+namespace Vaas
+{
+    public class Vaas

```

Comment:
Add

/// <remarks>not thread-safe</remarks>

or make thread-safe. This would require changes (all properties are not thread-safe). Contact me for more infos.

Reply:
That's a good call. General question: Does it need to be thread-safe? Only SDK I'm sure that is thread-safe is currently the Rust one. Either we make the C# version thread-safe or we expect the user to create a new VaaS object per thread. Any case, needs to be documented.

Reply:
Thread-safe now